l337-postgres/src/lib.rs

```diff
@@ -98,16 +98,19 @@ where
             return true;
         }
 
-        match conn.receiver.poll() {
+        // Use try_recv() as `has_broken` can be called via Drop
+        // and not have a future Context to poll on.
+        // https://docs.rs/futures/0.1.28/futures/sync/oneshot/struct.Receiver.html#method.try_recv
+        match conn.receiver.try_recv() {
```

Comment:
Hm, try_recv doesn't schedule a wakeup for this future though.

I suppose this is ok though, as we manually call `has_broken` on every put back. Did you come to the same conclusion here @DarrenTsung?

Reply:
Looking at the code more it seems we were always throwing away the wake event anyways. The only time we checked this was on `put_back`, so not scheduling a wake event shouldn't matter here.

Reply:
Yeah not sure we need a wakeup event since we're not waiting on the receiver